### PR TITLE
Disambiguate horizontal and vertical scrolling

### PR DESCRIPTION
**Pull request by abarth**
We now have separate gestures for horizontal, vertical, and pan scrolling.

### CODE REVIEW

File: sky/packages/sky/lib/gestures/scroll.dart [hunk 1]
```diff
@@ -15,43 +15,44 @@ enum ScrollState {
...
+  GestureScrollStartCallback onStart;
+  Function onUpdate;
+  GestureScrollEndCallback onEnd;
 
-  ScrollState state = ScrollState.ready;
-  sky.Offset pendingScrollOffset;
+  ScrollState _state = ScrollState.ready;
+  dynamic _pendingScrollDelta;
+
+  dynamic get _initialPendingScrollDelta;
```
Comment (Hixie) [inline sky/packages/sky/lib/gestures/scroll.dart]: use a generic

**Comment by Hixie** [comment]
LGTM. This is great.

**Comment by eseidel** [comment]
Is this applied to the widgets yet?  Does the card demo feel better, for instance?

**Comment by abarth** [comment]
This is applied to widgets.  It works in PinkSky.  Hans is working integrating it with dismissable.

**Pull request merged**